src/TopSurface.cpp

```diff
@@ -37,14 +37,28 @@ bool TopSurface::ironing(const SliceMeshStorage& mesh, const GCodePathConfig& li
     const bool zig_zaggify_infill = pattern == EFillMethod::ZIG_ZAG;
     constexpr bool connect_polygons = false; // midway connections can make the surface less smooth
     const coord_t line_spacing = mesh.settings.get<coord_t>("ironing_line_spacing");
-    const coord_t outline_offset = -mesh.settings.get<coord_t>("ironing_inset");
     const coord_t line_width = line_config.getLineWidth();
     const std::vector<AngleDegrees>& top_most_skin_angles = (mesh.settings.get<size_t>("roofing_layer_count") > 0) ? mesh.roofing_angles : mesh.skin_angles;
     assert(top_most_skin_angles.size() > 0);
     const AngleDegrees direction = top_most_skin_angles[layer.getLayerNr() % top_most_skin_angles.size()] + AngleDegrees(90.0); //Always perpendicular to the skin lines.
     constexpr coord_t infill_overlap = 0;
     constexpr int infill_multiplier = 1;
     constexpr coord_t shift = 0;
+
+    coord_t ironing_inset = -mesh.settings.get<coord_t>("ironing_inset");
+    if (pattern == EFillMethod::ZIG_ZAG && ironing_inset == 0)
+    {
+        //Compensate for the outline_offset decrease that takes place when using the infill generator to generate ironing with the zigzag pattern
+        const float width_scale = (float)mesh.settings.get<coord_t>("layer_height") / mesh.settings.get<coord_t>("infill_sparse_thickness");
+        ironing_inset += width_scale * line_width / 2;
```

Comment:
Somehow this doesn't seem to be working correctly. I tested this with a 1cm cube centered around the origin, and with the Ironing Inset at 0, and I got the following g-code to iron it:

```
;TYPE:SKIN
G1 F1500 E111.37361
G1 F1200 X4.999 Y-4.899 E111.37369
G1 X4.999 Y-4.758 E111.37378
G1 X4.758 Y-4.999 E111.374
G1 X4.616 Y-4.999 E111.37409
G1 X4.999 Y-4.616 E111.37443
G1 X4.999 Y-4.475 E111.37451
G1 X4.475 Y-4.999 E111.37498
G1 X4.333 Y-4.999 E111.37507
G1 X4.999 Y-4.333 E111.37566
G1 X4.999 Y-4.192 E111.37575
G1 X4.192 Y-4.999 E111.37646
```
[Zipped project file](https://github.com/Ultimaker/CuraEngine/files/4616309/iron_around_centre.zip)

You can see that it goes to X4.999 and Y-4.999. Bar a rounding error of 1 micron, this is exactly to the border with the centreline of the ironing line.

I'd expect that it goes to the edge of the top surface with the edge of the ironing line. In that project file the Skin Line Width is 0.4mm and the ironing flow is 10% of that, so it should keep 0.04mm away from the border.

That is, unless we consider this the new expected behaviour.

Reply:
I did not take into account the ironing flow. Fixed in https://github.com/Ultimaker/CuraEngine/pull/1251/commits/b49ddf7c5c73c76c5a6aac788cb8acafeef60794 by moving the ironing line by ironing_line_width/2 inwards.